magnolia-voting-app/src/main/java/com/example/votingapp/templatingfunctions/PollReportTemplatingFunctions.java

```diff
@@ -0,0 +1,184 @@
+package com.example.votingapp.templatingfunctions;
+
+import info.magnolia.cms.security.JCRSessionOp;
+import info.magnolia.context.MgnlContext;
+import info.magnolia.jcr.util.NodeUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PollReportTemplatingFunctions {
+
+    private static final Logger log = LoggerFactory.getLogger(PollReportTemplatingFunctions.class);
+
+    private static String POLLS = "polls";
+    private static String SUM_QUESTIONS_NUMBER = "sumQuestionsNumber";
+    private static String SUM_ANSWERS_NUMBER = "sumAnswersNumber";
+    private static String SUM_VOTES_NUMBER = "sumVoteNumber";
+
+    Map<String, Integer> contentNodeMap = new HashMap<>();
+    Map<String, Integer> answerNodeMap = new HashMap<>();
+    Map root = new HashMap<>();
+
+    List<Node> level3 = new ArrayList<>();
```

Comment:
It should have meaningful name, I don't know what level is. I guess that you will take time to remember what you have done after months. Keep in mind that coding is not just for running, also about maintenance costs. 

getAllChildren should return list of child, but only string = null is return, consider to rename method to process something with void method.

I hope that Node will reflex as an object in the future, which have very clear structure. List or map here is hard to maintain.